app/misc.py

```diff
@@ -165,6 +166,36 @@ def is_mod(self, sid, power_level=2):
         """ Returns True if the current user is a mod of 'sub' """
         return is_sub_mod(self.uid, sid, power_level, self.can_admin)
 
+    @cache.memoize(1)
+    def mod_notifications(self):
+        if self.is_mod:
+            post_report_counts = (
+                SubPostReport.select(Sub.sid, fn.COUNT(SubPostReport.id).
+                                     alias('count')).
+                join(SubPost).join(Sub).join(SubMod).
+                where((SubMod.user == self.uid) & SubPostReport.open).
+                group_by(Sub.sid).dicts())
+            comment_report_counts = (
+                SubPostCommentReport.select(Sub.sid,
+                                            fn.COUNT(SubPostCommentReport.id).
+                                            alias('count')).
+                    join(SubPostComment).join(SubPost).join(Sub).join(SubMod).
+                    where((SubMod.user == self.uid) & SubPostCommentReport.open).
+                    group_by(Sub.sid).dicts())
+
+            counts = defaultdict(int)
+            for item in post_report_counts:
+                counts[item['sid']] = item['count']
+            for item in comment_report_counts:
+                counts[item['sid']] += item['count']
+
+            return [[sid, count] for sid, count in counts.items()]
```

Comment:
This function doesn't seem to account for the special case of administrators and default subs. Admins are assumed to moderate all default subs (see `is_sub_mod`), but this function won't count pending notifications for default subs if the user is an admin.

Reply:
After double checking, `/mod/reports` doesn't seem to comply with this assumption either so this might be material for a separate ticket

Reply:
I deliberately didn’t include all the subs in the count for admins so that admins could see the count of open reports that they have to deal with in the subs they moderate themselves.

Reply:
Nvm it dawned on me what you are talking about. Should we maybe make SubMod entries for the default subs instead of special-casing them?